internal/observability/call.go

```diff
@@ -496,6 +499,17 @@ func (c call) logStreamEvent(err error, success bool, succMsg, errMsg string, ex
 	ce.Write(fields...)
 }
 
+// emitSpanErrorTag sets the error information as tags on the current span
+func (c call) emitSpanErrorTag(res callResult) {
+	if span := opentracing.SpanFromContext(c.ctx); span != nil {
```

Comment:
My concern of using obs middleware is that the required inbound/ outbound spans are not available in go context when you are trying to obtain spans here.

Could you double check?

Reply:
After reviewing the code more thoroughly, I believe you're right. It doesn’t seem straightforward to pass the original spans here. Do you have any suggestions on how we might approach this?